e2e_test/sink/kafka/create_sink.slt

```diff
@@ -183,6 +183,50 @@ create sink invalid_pk_column from t_kafka with (
     primary_key = 'id,invalid'
 );
 
+### Test sink with key encode ###
+
+statement error sink key encode unsupported: JSON, only TEXT supported
+create sink sink_text_error from t_kafka with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-rw-sink-text-key-id',
+  primary_key = 'id')
+format plain encode json (
+  force_append_only='true'
+) key encode json ;
+
+statement error The key encode is TEXT, but the primary key has 2 columns. The key encode TEXT requires the primary key to be a single column.
+create sink sink_text_error from t_kafka with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-rw-sink-text-key-id',
+  primary_key = 'id, v_varchar')
+format plain encode json (
+  force_append_only='true'
+) key encode text ;
+
+statement error The key encode is TEXT, but the primary key column v_bytea has type bytea. The key encode TEXT requires the primary key column to be of type varchar, bool, small int, int, or big int.
+create sink sink_text_error from t_kafka with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-rw-sink-text-key-id',
+  primary_key = 'v_bytea')
+format plain encode json (
+  force_append_only='true'
+) key encode text ;
+
+statement ok
+create sink sink_text_id from t_kafka with (
+  connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+  topic = 'test-rw-sink-text-key-id',
+  primary_key = 'id')
+format plain encode json (
+  force_append_only='true'
+) key encode text ;
```

Comment:
Ingest it back with `include key as` and compare it with expected value?

Reply:
already included in the test.